dev-client/src/components/content/typography/DisableableText.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {Text} from 'terraso-mobile-client/components/NativeBaseAdapters';
+
+type DisableableTextProps = {
+  disabled?: boolean;
+} & React.PropsWithChildren;
+
+export const DisableableText = ({
+  disabled = false,
+  children,
+}: DisableableTextProps) => {
+  return <Text color={disabled ? 'text.disabled' : undefined}>{children}</Text>;
```

Comment:
suggestion: style disabled text in theme.ts instead of with an inline color here.

Reply:
I don't know if it's correct to make a full variant for disabled text; it's an attribute we're adding to other text, not a first-class type like a heading, paragraph etc. The notion of "disabled" text is fully encapsulated by this component and it already uses the theme for the color key itself.

Reply:
You don't need to make a variant for that. React Native already supports `_disabled` as a sub-style. We've done this in other places in theme.ts.

Reply:
The `disabled` property doesn't work for text the way it does for e.g. buttons; if a `_disabled` section is specified in the theme for text as a base style or in variants it doesn't get triggered by setting its `disabled` property. I really think that this is the simplest solution. I will also note that `theme.ts` as a whole is a NativeBase construct which we are trying to move away from.